### DISCUSSION TITLE
Set Redirect URL (Running Twill On GitHub Codespaces)

### DISCUSSION THREAD

**Discussion by fweth** [area17/twill#2370] [General]
Anybody managed to get Twill (basic page builder) running in Codespaces?  I followed the instructions and the situation is as follows:

- `https://xxx-xxx-xxx-8000.app.github.dev/` is my `localhost`, i.e. the URL under which I can reach the port
- On this URL, it shows the error `Call to a member function isNotEmpty() on null`, I think this is as expected, because I haven't created an index page yet
- `https://xxx-xxx-xxx-8000.app.github.dev/admin` redirects to `https://xxx-xxx-xxx-8000.app.github.dev:8000/admin/login` which yields 404
- Manually changing to `https://xxx-xxx-xxx-8000.app.github.dev/admin/login` successfully shows me the Twill login page
- After entering my login data, it tries to redirect to `localhost`, which doesn't exist of course.
- I tried already to set `APP_URL` and `ADMIN_APP_URL` to my GitHub URL, but this doesn't seem to affect the redirect after authentification.
- Any idea where I can set the redirect URL?

**Comment by Zana12** [comment]
I have the same issue running Twill in Codespaces. I have tried it even without the basic-page-builder. Even though the same problem with the redirection to `https://xxx-xxx-xxx-8000.app.github.dev:8000/` or `http://localhost:8000/`. Not even clearing config or routes helps at this point.
